### PR TITLE
#28/logic/element creator oop

### LINKED ISSUE #28: Logika | Kreator elementów DOM i dodawanie 'dzieci'
### Kryteria akceptacji
**Należy zweryfikować i skonsultować, a także ewentualnie uaktualnić przed rozpoczęciem prac nad zgłoszeniem.**

- [ ] Funkcja pozwalająca tworzyć elementy DOM za pomocą funkcji
- [ ] Funkcja pozwalająca dodawać 'dzieci' do elementu 'rodzica' w drzewie DOM

### PR DESCRIPTION
**Pull request by Smyku6**
closes #28

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,8 @@
   ],
   "scripts": {
     "start": "parcel",
-    "build": "parcel build"
+    "build": "parcel build",
+    "clean": "rimraf .cache && rimraf coverage && rimraf dist && rimraf node_modules"
```
Comment (vieraboschkova) [inline package.json:12]: coverage is a directory connected to jest and testing, I think you can remove it

File: src/app/controller/controller.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    // RENDER LOGO
+    this.view.render('.header__game-logo', createLogo());
+    // RENDER GAME MODE BUTTONS
+    this.view.render('.header__game-nav', ...gameModeButtons);
+    // RENDER INIT LOOK OF GAME_MODE
+    this.view.render('.game__mode', 'Choose category');
+    // RENDER RULES
+    this.view.render('.game__mode-rules', 'Harry Potter Quiz Game');
+  }
```
Comment (vieraboschkova) [inline src/app/controller/controller.js:23]: The controller doesnt need to know all that. Perhaps you could create a view function renderInitialScreen which will do all these things

File: src/app/controller/controller.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  // Funkcja znajduje się w kontrolerze, odpalana jest w momencie kliknięcia buttona na stronie.
+  // Po kliknięciu kontroler:
+  // a) zmienia wartość gameMode w obiekcie model
+  // b) używa metody showQuestionForChoosenMode, żeby zaktualizować tekst w odpowiednim
+  // elemencie DOM
+  gameModeChoosen(mode) {
```
Comment (vieraboschkova) [inline src/app/controller/controller.js:30]: Maybe you can call it starting with a verb, something like switchGameMode, changeGameMode?

File: src/app/view/components/createLogo.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import elementCreator from '../../utils/elementCreator';
+import logoPath from '../../../../static/assets/img/logo.png';
+
+const createLogo = () => {
```
Comment (vieraboschkova) [inline src/app/view/components/createLogo.js:4]: I think this content is also from another PR. I think it shouldn't be here also

File: package.json (hunk 1 above)
Comment (Smyku6) [reply package.json:12]: fixed

File: index.html [hunk 5]
```diff
@@ -15,31 +15,37 @@
...
+            </div>
+            <div class="header__game-nav">
+
+            </div>
         </header>
 
         <aside class="game-image">
-            <div class="game-image__content"></div>
+            <div class="game-image__content">
+                <img src="./static/assets/img/logo.png" alt="jajko">
```
[the remark this answers is not in the document]
Comment (Smyku6) [reply index.html]: 🥚🥚🥚

File: src/app/view/components/gameModeButtons.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import elementCreator from '../../utils/elementCreator';
+
```
Comment (vieraboschkova) [inline src/app/view/components/gameModeButtons.js:2]: I think this file is not necessary in this PR ?

File: src/app/logic/localStorage.js [hunk 7]
```diff
@@ -1,9 +1,25 @@
-const testObject = {
-  name: 'Albus Dumbledore',
-  result: 200,
-  mode: 'Characters',
-};
-
-function _saveGameInfoInLocalStorage(object) {
-  localStorage.setItem('gameinfo', JSON.stringify(object));
+function addResult(rankings, gameInfo, amountOfResults) {
```
[the remark this answers is not in the document]
Comment (Smyku6) [reply src/app/logic/localStorage.js:1]: deleted

File: src/app/view/components/createLogo.js (hunk 4 above)
Comment (Smyku6) [reply src/app/view/components/createLogo.js:4]: deleted

File: src/app/view/components/gameModeButtons.js (hunk 6 above)
Comment (Smyku6) [reply src/app/view/components/gameModeButtons.js:2]: deleted

**Review by vieraboschkova — CHANGES_REQUESTED** [review]
You did a lot of work! Thanks a lot, I think the team will see the project more clearly now!
Clean the files that are out of the scope of your task and you're good to go!

**Review by vieraboschkova — APPROVED** [review]
Great! Thanks! Amazing job!

**Pull request merged**